### ISSUE TITLE
Edit in OneDrive fails

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/HPI#1125]
Looks like something is broken with the OneDrive integration.  When I try to do an edit online from edge or release, I see this open in a new tab:

![image](https://cloud.githubusercontent.com/assets/1377228/16278886/3302fee6-387e-11e6-9359-d8e7343b5812.png)

In case anyone needs to copy paste from the error message:

```
{"error":{"code":"generalException","message":"General Exception While Processing","innererror":{"errorType":"server","debugMessage":"GeneralStorageException(GeneralException): The input is not a valid Base-64 string as it contains a non-base 64 character, more than two padding characters, or an illegal character among the padding characters. \r\n","stackTrace":"   at System.Convert.FromBase64_Decode(Char* startInputPtr, Int32 inputLength, Byte* startDestPtr, Int32 destLength)\r\n   at System.Convert.FromBase64CharPtr(Char* inputPtr, Int32 inputLength)\r\n   at System.Convert.FromBase64String(String s)\r\n   at Microsoft.MSN.Storage.Api.CoreApi.Helpers.SharesHelper.GetKeySegmentsForShareId(String sharesKey) in d:\\dbs\\sh\\odht\\0619_181036\\cmd\\21\\server\\storage\\private\\api\\coreapi\\Helpers\\SharesHelper.cs:line 359\r\n   at Microsoft.MSN.Storage.Api.CoreApi.Helpers.SharesHelper.GetKeySegmentsForObfuscatedSharesKey(String sharesKey) in d:\\dbs\\sh\\odht\\0619_181036\\cmd\\21\\server\\storage\\private\\api\\coreapi\\Helpers\\SharesHelper.cs:line 104\r\n   at Microsoft.MSN.Storage.Api.CoreApi.Controllers.ShortUrlController.<GenerateRedirectResult>d__13.MoveNext() in d:\\dbs\\sh\\odht\\0619_181036\\cmd\\21\\server\\storage\\private\\api\\coreapi\\Controllers\\ShortUrlController.cs:line 147\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.MSN.Storage.Api.CoreApi.Controllers.ShortUrlController.<GetWithType>d__12.MoveNext() in d:\\dbs\\sh\\odht\\0619_181036\\cmd\\21\\server\\storage\\private\\api\\coreapi\\Controllers\\ShortUrlController.cs:line 63\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Threading.Tasks.TaskHelpersExtensions.<CastToObject>d__3`1.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Controllers.ApiControllerActionInvoker.<InvokeActionAsyncCore>d__0.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.MSN.Storage.Api.CoreApi.Filters.LoggingFilterAttribute.<ExecuteActionFilterAsync>d__0.MoveNext() in d:\\dbs\\sh\\odht\\0619_181036\\cmd\\21\\server\\storage\\private\\api\\coreapi\\Filters\\LoggingFilterAttribute.cs:line 59\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Controllers.ActionFilterResult.<ExecuteAsync>d__2.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.Dispatcher.HttpControllerDispatcher.<SendAsync>d__1.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Web.Http.Dispatcher.HttpControllerDispatcher.<SendAsync>d__1.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at System.Web.Http.HttpServer.<SendAsync>d__0.MoveNext()\r\n\r\n\r\n\r\n","throwSite":"1891.687b","code":"generalException"}}}
```

**Comment by ZachGerth** [comment]
It looks like a handful of parts of the API have changed since this was initially implemented - it's possible it expects different (or new) information in the URL, or we'll have to update parts that we've configured from the OneDrive/HPI side.

**Comment by rosenoam** [comment]
Fixed with OC commit: 15491
CR: Avi

**Comment by benallenallen** [comment]
It's baaaaaack. Looks like our string parsing is close, but adding on something extra at the end of our URL that we are building to launch OneDrive:

![image](https://cloud.githubusercontent.com/assets/4850976/21826488/4c42979a-d74d-11e6-8635-4bccd74cc56f.png)


If we remove the `","file":%7B"hashes":%7B"cr` from the URL and just hit that URL instead, it works beautifully, so I'm guessing we just have some code that is grabbing some extra bits from the json that we don't want to include in the url.

**Comment by rosenoam** [comment]
Fixed with OC Revision #15809
